lib/avromatic/model/attributes.rb

```diff
@@ -44,8 +44,22 @@ def required?
 
         def coerce(input)
           type.coerce(input)
+        rescue Avromatic::Model::UnknownAttributeError => e
+          raise Avromatic::Model::CoercionError.new("Value for #{owner.name}##{name} could not be coerced to a #{type.name} " \
+            "because the following unexpected attributes were provided: #{e.attributes.join(', ')}. " \
+            "Provided argument: #{input.inspect}")
```

Comment:
See comment below - it would be helpful to print the expected attributes after the unexpected ones, before the provided argument.